pyat/at/physics/dynamic_aperture.py

```diff
@@ -0,0 +1,1012 @@
+                self.dict_def_range[p][0] = coord[p]-step
+                if self.verbose:
+                    print('Computed range for {pl}: [{mi}, {ma}]'.format(pl=p,
+                                                                         mi=self.dict_def_range[p][0],
+                                                                         ma=self.dict_def_range[p][1]))
+
+        # restore initial parallel computation value
+        self.parallel_computation = init_parcomp
+
+        pass
+
+    def test_points_grid(self, compute_limits=False):
+        """
+        define fixed grid of test points
+        the grid of points is stored in a dictionary of coordinates, one array for each dimension
+
+        :param compute_limits calls compute_range to deterine the maximum ranges before defining the grid
+
+        :return:
+        """
+
+        if compute_limits:
+            self.compute_range()
+
+        # define grid
+        if self.grid_mode == 'radial':
+
+            d_ = {'x': [], 'xp': [], 'y': [], 'yp': [], 'delta': [], 'ct': []}
+
+            # find n_points more than 1
+            two_planes = [p for p in self.planes if self.n_points[p] > 1]
+
+            if len(two_planes) != 2:
+                raise IndexError('There are less or more than 2 planes with more than 1 point to scan')
+
+            ellipse_axes = [self.dict_def_range[two_planes[0]][1],
+                            self.dict_def_range[two_planes[1]][1]]
+
+            if self.dict_def_range[two_planes[1]][0] == 0:
+                theta = np.linspace(0, math.pi, self.n_points[two_planes[0]])
+            else:
+                theta = np.linspace(0, 2*math.pi, self.n_points[two_planes[0]])
+
+            ip = 0
+
+            for ind_p, p in enumerate(self.planes):  # must loop all plane to define all 6D ranges (1 point)
+
+                if self.n_points[p] > 1:  # if here, there are only 2 planes available
+
+                    # scan predefined range
+                    for ea in np.sqrt(np.linspace(ellipse_axes[ip]**2/self.n_points[two_planes[1]],
+                                          ellipse_axes[ip]**2, self.n_points[two_planes[1]])):
+                        if ip == 0:
+                            d_[p].append([ea * math.cos(t) for t in theta])
+                        elif ip == 1:
+                            d_[p].append([ea * math.sin(t) for t in theta])
+
+                    ip += 1
+
+                else:  # this dimension is not part of the scan
+                    d_[p].append([(self.dict_def_range[p][0] + self.dict_def_range[p][1]) / 2])
+
+            flat_dd = []
+            for k, val in d_.items():
+                flat_dd.append([item for sublist in val for item in sublist])
+                print('{} has {} elements'.format(k, len(flat_dd[-1])))
+
+            num_points = max([len(dd) for dd in flat_dd])
+
+            # define lists for DA scan
+            for ip, p in enumerate(self.planes):
```

Comment:
`ip` is not used:
`for p in self.planes:`

Reply:
done